### ISSUE TITLE
Default `bucket_size` in piped mode to 0

### ISSUE THREAD

**Issue by TimerErTim** [bensadeh/tailspin#109]
### Problem

I have debugged my setup for 2 hours because tailspin did not show my logs in piped mode. Only once I looked at the source code I realized, I have to set the bucket size to zero in order to show realtime log output. 

I expected this to be the default when using `tspin` in piped mode because with a bucket_size of 10,000, the output is delayed af. 

### Fix

Set the bucket size to zero by default when used in piped mode. This would prevent any future users from having the same problem. The `tspin` process runs in parallel anyway when used in pipes, so the performance loss should be neglectable. If more performance at the cost of "up-to-dateness" is required, then one could just manually increase the bucket size.

**Comment by guillenotfound** [comment]
Faced this issue today, the way other utils work like grep is using :

```
--line-buffered
             Force output to be line buffered.  By default, output is line buffered when standard output is a terminal and block buffered otherwise.
```

IMO it would be ideal to have the same param name if possible 👍

**Comment by bensadeh** [comment]
Thanks for letting me know, and sorry about that! I have changed the behavior or processing data from `stdin` to be sequential until there's a better solution in place.

Fixed in `2.4.0`.
